core/src/test/java/org/elasticsearch/index/IndexSettingsTests.java

```diff
@@ -370,6 +370,27 @@ public void testTranslogFlushSizeThreshold() {
         assertEquals(actualNewTranslogFlushThresholdSize, settings.getFlushThresholdSize());
     }
 
+    public void testTranslogGenerationSizeThreshold() {
+        final ByteSizeValue size = new ByteSizeValue(Math.abs(randomInt()));
```

Comment:
should we protect against 0? Thinking about it more, we should at least fit one op in there. So I wonder if we should have a decent lower bound. The problem in making it too big is testing - how about making it 1kb?